proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,807 @@
+# Machine readable JSON output for dotnet list package
+
+* Status: **In Review**
+* Author: [Erick Yondon](https://github.com/erdembayar)
+* GitHub Issue [7752](https://github.com/NuGet/Home/issues/7752)
+
+## Problem background
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many organizations are required by [regulation](https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/) to audit packages that they're using in a repository.
+
+Currently there's no easy way to produce a [Software Bill of Material (SBOM)](https://blog.sonatype.com/what-is-a-software-bill-of-materials) output which can be consumed by another auditing system or kept for records.
+
+* Parse-friendly output. Other package managers like [NPM already have it](https://docs.npmjs.com/cli/v7/commands/npm-ls) (`npm ls --parseable` and `npm ls --json`).
+* Useful for CI/CD auditing(compliance, security ..)
+  * Produce Software Bill of Material (SBOM) (compliance, historic keeping)
+  * Enhancing Software Supply Chain Security
+    * Check any vulnerable packages
+    * Check any deprecated packages
+    * Check any outdated packages
+  * Check license compliance
+  * Resolve dependency issues (detect duplicate packages, newly introduced dependencies, dependency removals, etc.)
+  * Making the output machine-readable unlocks additional tooling and automation scenarios in CI/CD pipeline above scenarios.
+
+## Who are the customers
+
+Anyone (government/private enterprises, security experts, individual contributors) who wants to consume `dotnet list package` output for auditing tool or historic keeping, CI/CD orchestrating.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+#### `--format` option
```

Comment:
`--format` is already used in `dotnet nuget list source`, though with a different set of valid arguments:
* Detailed
* Short

I think it might be confusing to the user to have both of these distinct sets of meaning, especially as I expect `--format` to become a thing across more SDK commands, so in the future we should look towards having an different way to get the detailed/short forms of `dotnet nuget list source` if those different forms are useful.

Reply:
Thank you for your feedback. I didn't even know there was formatting option for `dotnet nuget list source`. Difference between `Detailed` and `Short` output is trivial, maybe we can change it the future, I'll create tracking issue.

Reply:
https://github.com/nuget/home/issues/11615